### PR TITLE
CA-201119: Updating XenModel/XenAPI with the latest SDK

### PR DESCRIPTION
**Pull request by MihaelaStoica**
This commit contains fixes for CA-201119, CA-201361 and CA-202268

Signed-off-by: Mihaela Stoica <mihaela.stoica@citrix.com>

### CODE REVIEW

**Comment by stephen-turner** [comment]
@MihaelaStoica,

Two errors in FriendlyErrorNames have been deleted. Do we not still need them for old servers?

**Comment by MihaelaStoica** [comment]
@stephen-turner, Both errors that have been removed only existed in trunk, not in any released version.

File: XenModel/XenAPI/FriendlyErrorNames.resx [hunk 1]
```diff
@@ -136,7 +136,7 @@
     <value>Could not enable external authentication.</value>
   </data>
   <data name="AUTH_ENABLE_FAILED_DOMAIN_LOOKUP_FAILED" xml:space="preserve">
-    <value>XenServer was unable to contact your domain server to enable external authentication. Check that your settings are correct and a route to the server exists.</value>
+    <value>Unable to contact your domain server to enable external authentication. Check that your settings are correct and a route to the server exists.</value>
```
Comment (stephen-turner) [inline XenModel/XenAPI/FriendlyErrorNames.resx]: The server was unable

File: XenModel/XenAPI/FriendlyErrorNames.resx [hunk 2]
```diff
@@ -598,7 +598,7 @@
     <value>Your license has expired</value>
   </data>
   <data name="LICENSE_FILE_DEPRECATED" xml:space="preserve">
-    <value>This type of license file is for previous versions of XenServer. Please upgrade to the new licensing system.</value>
+    <value>This type of license file is for previous versions. Please upgrade to the new licensing system.</value>
```
Comment (stephen-turner) [inline XenModel/XenAPI/FriendlyErrorNames.resx]: previous versions of the server

File: XenModel/XenAPI/FriendlyErrorNames.resx [hunk 3]
```diff
@@ -607,7 +607,7 @@
     <value>There was an error processing your license.  Please contact your support representative</value>
   </data>
   <data name="LIKEWISE_WINERROR_0251E" xml:space="preserve">
-    <value>XenServer was unable to contact your domain server. Check your settings are correct and that a route to the server exists.</value>
+    <value>Unable to contact your domain server. Check your settings are correct and that a route to the server exists.</value>
```
Comment (stephen-turner) [inline XenModel/XenAPI/FriendlyErrorNames.resx]: The server was unable

File: XenModel/XenAPI/FriendlyErrorNames.resx [hunk 4]
```diff
@@ -718,7 +718,7 @@
     <value>The specified patch is applied and cannot be destroyed.</value>
   </data>
   <data name="PATCH_PRECHECK_FAILED_ISO_MOUNTED" xml:space="preserve">
-    <value>XenServer Tools ISO must be ejected from all running VMs in the pool.</value>
+    <value>Tools ISO must be ejected from all running VMs in the pool.</value>
```
Comment (stephen-turner) [inline XenModel/XenAPI/FriendlyErrorNames.resx]: The Tools ISO

File: XenModel/XenAPI/FriendlyErrorNames.resx [hunk 5]
```diff
@@ -733,10 +733,10 @@
     <value>The update precheck stage failed: there are one or more VMs still running on the server.  All VMs must be shut down or suspended before the update can be applied.</value>
   </data>
   <data name="PATCH_PRECHECK_FAILED_WRONG_SERVER_BUILD" xml:space="preserve">
-    <value>XenServer build number ({1}) is incompatible with the update build number ({2}).</value>
+    <value>Current build number ({1}) is incompatible with the update build number ({2}).</value>
```
Comment (stephen-turner) [inline XenModel/XenAPI/FriendlyErrorNames.resx]: The current build number

File: XenModel/XenAPI/FriendlyErrorNames.resx [hunk 6]
```diff
@@ -733,10 +733,10 @@
     <value>The update precheck stage failed: there are one or more VMs still running on the server.  All VMs must be shut down or suspended before the update can be applied.</value>
   </data>
   <data name="PATCH_PRECHECK_FAILED_WRONG_SERVER_BUILD" xml:space="preserve">
-    <value>XenServer build number ({1}) is incompatible with the update build number ({2}).</value>
+    <value>Current build number ({1}) is incompatible with the update build number ({2}).</value>
   </data>
   <data name="PATCH_PRECHECK_FAILED_WRONG_SERVER_VERSION" xml:space="preserve">
-    <value>XenServer version number ({1}) is incompatible with the update version number regular expression ({2}).</value>
+    <value>Current version number ({1}) is incompatible with the update version number regular expression ({2}).</value>
```
Comment (stephen-turner) [inline XenModel/XenAPI/FriendlyErrorNames.resx]: The current build number

File: XenModel/XenAPI/FriendlyErrorNames.resx [hunk 7]
```diff
@@ -817,7 +820,7 @@
     <value>Could not enable external authentication: {1}</value>
   </data>
   <data name="POOL_AUTH_ENABLE_FAILED_DOMAIN_LOOKUP_FAILED" xml:space="preserve">
-    <value>XenServer was unable to contact your domain server to enable external authentication. Check that your settings are correct and a route to the server exists.</value>
+    <value>Unable to contact your domain server to enable external authentication. Check that your settings are correct and a route to the server exists.</value>
```
Comment (stephen-turner) [inline XenModel/XenAPI/FriendlyErrorNames.resx]: The server was unable

File: XenModel/XenAPI/FriendlyErrorNames.resx [hunk 8]
```diff
@@ -870,7 +876,7 @@ Authorized Roles: {1}</value>
     <value>Cannot restore on this server because it was saved on an incompatible version</value>
   </data>
   <data name="RESTORE_SCRIPT_FAILED" xml:space="preserve">
-    <value>The restore could not be performed because the restore script failed. Your XenServer Backup file may be corrupt. See the logs for more information.</value>
+    <value>The restore could not be performed because the restore script failed. Your Backup file may be corrupt. See the logs for more information.</value>
```
Comment (stephen-turner) [inline XenModel/XenAPI/FriendlyErrorNames.resx]: backup (lower case)

File: XenModel/XenAPI/FriendlyErrorNames.resx [hunk 9]
```diff
@@ -1478,12 +1484,12 @@ Authorized Roles: {1}</value>
...
     <value>Only 1 LUN may be used with shared LVM</value>
   </data>
-  <data name="SR_DETACHED_ON_MASTER" xml:space="preserve">
-    <value>The SR is currently detached on the master.</value>
-  </data>
   <data name="SR_DEVICE_IN_USE" xml:space="preserve">
     <value>The SR operation cannot be performed because a device underlying the SR is in use by the server.</value>
   </data>
+  <data name="SR_DOES_NOT_SUPPORT_MIGRATION" xml:space="preserve">
+    <value>You attempted to migrate a VDI on SR which doesn't have snapshot capability</value>
```
Comment (stephen-turner) [inline XenModel/XenAPI/FriendlyErrorNames.resx]: I don't understand this message. Is it onto an SR or from an SR. Also, what have snapshots got to do with it?

File: XenModel/XenAPI/FriendlyErrorNames.resx [hunk 10]
```diff
@@ -1676,6 +1682,9 @@ Authorized Roles: {1}</value>
   <data name="VIF_IN_USE" xml:space="preserve">
     <value>Network has active VIFs</value>
   </data>
+  <data name="VIF_NOT_IN_MAP" xml:space="preserve">
+    <value>This VIF was not mapped to a destination Network in VM.migrate_send operation</value>
```
Comment (stephen-turner) [inline XenModel/XenAPI/FriendlyErrorNames.resx]: "to a destination network</value>" (lower case "network", and end the string there).

File: XenModel/XenAPI/FriendlyErrorNames.resx [hunk 11]
```diff
@@ -1875,7 +1884,7 @@ Authorized Roles: {1}</value>
     <value>WLB rejected the server's request as malformed.</value>
   </data>
   <data name="WLB_MALFORMED_RESPONSE" xml:space="preserve">
-    <value>The WLB server said something that XenServer wasn't expecting or didn't understand.</value>
+    <value>The WLB server said something that your server wasn't expecting or didn't understand.</value>
```
Comment (stephen-turner) [inline XenModel/XenAPI/FriendlyErrorNames.resx]: "the server", rather than "your server"

File: XenModel/XenAPI/FriendlyErrorNames.resx (hunk 9 above)
Comment (MihaelaStoica) [inline XenModel/XenAPI/FriendlyErrorNames.resx]: The message is used for both _onto_ and _from_ and it refers to the "vdi_snapshot" sm capability.
We decided to change it to "You attempted to migrate a VDI to or from an SR that doesn't support migration"

**Comment by MihaelaStoica** [comment]
I've updated the friendly error names in the xen-api-sdk repository with the suggestions above. Please see https://github.com/xapi-project/xen-api-sdk/pull/65.

**Comment by stephen-turner** [comment]
Thanks, merging now.

**Pull request merged**